pyomo/contrib/gdpopt/util.py

```diff
@@ -113,8 +115,45 @@ def presolve_lp_nlp(solve_data, config):
     return False, None
 
 
+def epigraph_reformulation(exp, slack_var_list, constraint_list, use_mcpp, sense):
+    """Epigraph reformulation.
+
+    Generate the epigraph reformuation for objective expressions.
+
+    Parameters
+    ----------
+    slack_var_list : VarList
+        Slack vars for epigraph reformulation.
+    constraint_list : ConstraintList
+        Epigraph constraint list.
+    use_mcpp : Bool
+        Whether to use mcpp to tighten the bound of slack variables.
+    exp : Expression
+        The expression to reformualte.
+    sense : objective sense
+        The objective sense.
+    """
+    slack_var = slack_var_list.add()
+    if mcpp_available() and use_mcpp:
```

Comment:
This is an open question, but should we have MCPP as the default and FBBT as the fallback? I get that both are contrib packages but recent PRs make the FBBT installation "easier" than MC++, and the previous version was completely included in Pyomo. Any thoughts @michaelbynum ?

Reply:
@bernalde In this context, the pure python version should work fine, which would not require any compiled extensions. If you already use MC++ or FBBT elsewhere, then that may influence your decision.

Reply:
By the way, the pure python implementation of FBBT will not go away (exactly for this reason - it can be cumbersome for some users to compile extensions). I expect the Appsi implementation to be a fast alternative to the pure python version. Also, in my experience, the compiled version is only faster if you need to use it iteratively.

Reply:
Well, this solves it. Let's use FBBT as default and MC++ as a fallback option.

Reply:
No problem.

Reply:
In MindtPy configurations, `use_mcpp = False` by default. Therefore, MindtPy already uses FBBT as default and MC++ as a fallback option. I have changed the default value of `use_mcpp` in `process_objective` function.